setup/install/providers/aws/aws-eks.md

```diff
@@ -0,0 +1,200 @@
+
+If you use the console to create the cluster, you must ensure that the same IAM user credentials are in the AWS SDK credential chain when you are running kubectl commands on your cluster.
+
+In the setup as done above, we used AWS CLI , hence you must ensure that the server/workstation from where you are running the following kubectl commands have the same AWS credentials.
+
+
+*  Create default kubectl configuration file
+
+Paste the following to your kubeconfig file , replace `<endpoint-url>` , `<base64-encoded-ca-cert>` and `<cluster-name>` with values of $EKS_CLUSTER_ENDPOINT , $EKS_CLUSTER_CA_DATA and $EKS_CLUSTER_NAME
+as noted above
+
+```yaml
+
+apiVersion: v1
+clusters:
+- cluster:
+    server: <endpoint-url>
+    certificate-authority-data: <base64-encoded-ca-cert>
+  name: kubernetes
+contexts:
+- context:
+    cluster: kubernetes
+    user: aws
+  name: aws
+current-context: aws
+kind: Config
+preferences: {}
+users:
+- name: aws
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1alpha1
+      command: heptio-authenticator-aws
+      args:
+        - "token"
+        - "-i"
+        - "<cluster-name>"
+        # - "-r"
+        # - "<role-arn>"
+      # env:
+        # - name: AWS_PROFILE
+        #   value: "<aws-profile>"
+
+```
+
+* Execute the following commands, which will create the necessary services accounts and cluster role bindings
+
+```
+CONTEXT=aws
+kubectl create namespace spinnaker
+#kubectl apply -f https://raw.githubusercontent.com/spinnaker/spinnaker.github.io/master/setup/install/providers/aws/eks/eks-admin-service-account.yaml 
+#kubectl apply -f https://raw.githubusercontent.com/spinnaker/spinnaker.github.io/master/setup/install/providers/aws/eks/eks-admin-cluster-role-binding.yaml 
+kubectl apply -f https://raw.githubusercontent.com/spinnaker/spinnaker.github.io/master/setup/install/providers/aws/eks/spinnaker-service-account.yaml 
+kubectl apply -f https://raw.githubusercontent.com/spinnaker/spinnaker.github.io/master/setup/install/providers/aws/eks/spinnaker-cluster-role-binding.yaml 
+TOKEN=$(kubectl get secret --context $CONTEXT \
+   $(kubectl get serviceaccount spinnaker-service-account \
+       --context $CONTEXT \
+       -n spinnaker \
+       -o jsonpath='{.secrets[0].name}') \
+   -n spinnaker \
+   -o jsonpath='{.data.token}' | base64 --decode)
+kubectl config set-credentials ${CONTEXT}-token-user --token $TOKEN
+kubectl config set-context $CONTEXT --user ${CONTEXT}-token-user
+```
+
+> This point onwards, the location (either EC2 instance, or container or local) where you run halyard commands from should have the kubeconfig file that was created
+by the above kubectl commands. If you ran the kubectl commands locally and want to run halyard command on an instance, then you can copy the kubeconfig file
+from local to the instance.
+
+## Enable Kubernetes Cloud provider using Halyard
+
+```
+hal config provider kubernetes enable
```

Comment:
These commands are duplicates of those in the top-level instructions: https://www.spinnaker.io/setup/install/providers/kubernetes-v2/#adding-an-account

These docs should ideally just explain what setup needs to be done for EKS so we avoid rewriting the same instructions in multiple places.

Reply:
IMO, it wont hurt to keep it here , so that users are following a logical flow when taking steps.